public/js/common/react_modal.js

```diff
@@ -0,0 +1,84 @@
+function openReactModal(courseCode) {
+    
+    var courseName = getCourseTitle(courseCode);
+    var formattedName = formatCourseName(courseCode);
+    var courseVideoUrls = getCourseVideoUrls(formattedName);
+    
+    React.render(
+        <ModalContent course={formattedName[0]}/>,
+        document.getElementById('modal-content-container')
+    );
+}
+
+var ModalContent = React.createClass({
+    render: function() {
+        return (
+            <div>
+                <Description course={this.props.course} />
+            </div>
+        );
+    }
+});
+
+var Description = React.createClass({
+  getInitialState: function() {
+    return {
+        course: [],
+        sessions: []
+    };
+  },
+
+  componentDidMount: function(){
+      //This loads the course json
+      $.ajax({
+        url: 'course',
+        data: {name: this.props.course},
+        dataType: 'json',
+        success: function(data) {
+            if (this.isMounted()) {
+                this.setState({course: data});
+                
+                console.log(data);
+                
+                //Not sure if there is a cleaner way to do this
+                //This is getting the session times
+                var lectures = data.fallSession.lectures
+                                         .concat(data.springSession.lectures)
+                                         .concat(data.yearSession.lectures);
+                var sessions = [];
+                $.each(lectures, function( index, value ) {
+                    sessions.push(value.code+value.session + "-" + value.section +": " + value.timeStr);
+                    console.log( value.code+value.session + "-" + value.section +": " + value.timeStr );
```

Comment:
`console.log` tends to be good for ad hoc developing purposes, but can mess up the developer console for programmers/users who don't know why something is being printed out. Are these necessary?